router/http.go

```diff
@@ -368,101 +395,122 @@ func (h *httpSyncHandler) Remove(id string) error {
 }
 
 func (s *HTTPListener) listenAndServe() error {
-	var err error
-	s.listener, err = listenFunc("tcp4", s.Addr)
-	if err != nil {
-		return listenErr{s.Addr, err}
-	}
-	if s.proxyProtocol {
-		s.listener = proxyproto.Listener{s.listener}
-	}
+	s.listeners = nil
```

Comment:
I can't remember exactly, but I remember it being a good reason. I'll do the explicit `Close()` first.